plugins/reminder.rb

```diff
@@ -0,0 +1,35 @@
+#  -*- coding: utf-8 -*-
+#  reminder.rb
+#  Author: slackR | slackErEhth77
+#  ------------------------
+#  A Cinch plugin to set reminder  for a given time.
+#  ------------------------
+#  This code is licensed by slackErEhth77 under the MIT License.
+#  http://opensource.org/licenses/MIT
+
+require_relative 'yossarian_plugin'
+
+	class Reminder < YossarianPlugin
+		include Cinch::Plugin
+		
+		def usage
+			"!remind <count> <unit> 'Message'."
+		end
+		
+		def match(cmd)
+			cmd =~ /^(!)?remind$/
+		end
+	
+	match /remind (\d+) (\w+) (.+)/, method: :set_reminder
+		
+		def convert_to_sec(count, unit)
+			count = (count*60 ) if unit =~ /min/
+			count = (count*3600) if unit =~ /hour/
+			return count
+		end
+		
+		def set_reminder(m, count, unit, message)
+			  count = convert_to_sec(count, unit) if unit =~ /min|hour/
```

Comment:
instead of putting the logic in `convert_to_sec`, just use a conditional block here